Add types for the WICG File and Directory Entries API

HenningCash opened the pull request:
So there are already types for the WICG File System Access API [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/wicg-file-system-access/index.d.ts). But, atm the more supported FS API in browsers is the "[File and Directory Entries API](https://wicg.github.io/entries-api/#dom-htmlinputelement-webkitentries)", even if it's no real standard.

Because @RReverser created `@types/wicg-file-system-access`, may I ask you for your opinion on this PR?

--- 

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

RReverser commented:
> Because @RReverser created `@types/wicg-file-system-access`, may I ask you for your opinion on this PR?

Made a quick glance, looks good, although I didn't carefully go and compare 1:1 with the spec - I trust you on this :)

RReverser commented on types/wicg-entries-api/index.d.ts:
```diff
@@ -0,0 +1,81 @@
+type FileCallback = (file: File) => void;
+
+interface FileSystemFlags {
+    readonly create: false;
```
Actually, a quick note: those properties should be optional.

RReverser commented on types/wicg-entries-api/index.d.ts:
```diff
@@ -0,0 +1,80 @@
+type FileCallback = (file: File) => void;
+
+interface FileSystemFlags {
+    readonly create?: false;
```
Hmm, also - why are they hardcoded to `false`? Shouldn't they accept any booleans?

HenningCash replied:
Hmm, not sure if I'm interpreting the specs correctly:
```
dictionary FileSystemFlags {
    boolean create = false;
    boolean exclusive = false;
};
```
Does that means it's a `boolean` that defaults to `false` or a `boolean` thats always `false`? :thinking:

HenningCash replied:
Not sure where I got that `readonly` from...

HenningCash commented:
Thank you for your quick feedback, RReverser! After a cup of coffee I re-checked the types and did some minor tweaks:
* Fixed the header-comment (not sure if the format is correct now)
* Removed readonly from `FileSystemFlags` props and removed the restriction to `false` (guess I  miss-interpreted the specs there)

RReverser commented:
> @RReverser: Can I add you as a second code-owner?

I guess, I assume there won't be many future changes on otherwise "dead" spec but why not :)

elibarzilay commented:
@HenningCash, do you plan to patch this by adding @RReverser as an owner?

RReverser commented on types/wicg-entries-api/index.d.ts:
```diff
@@ -0,0 +1,74 @@
+    }
+
+    interface FileSystemDirectoryReader {
+        readEntries(successCallback?: FileSystemEntriesCallback, errorCallback?: ErrorCallback): void;
```
Looking at https://wicg.github.io/entries-api/#api-directoryreader, first callback isn't optional.

```suggestion
        readEntries(successCallback: FileSystemEntriesCallback, errorCallback?: ErrorCallback): void;
```

RReverser commented on types/wicg-entries-api/index.d.ts:
```diff
@@ -0,0 +1,74 @@
+    }
+
+    interface FileSystemFileEntry extends BaseFileSystemEntry {
+        file(successCallback?: FileCallback, errorCallback?: ErrorCallback): void;
```
Same here (tests will need to be updated separately). https://wicg.github.io/entries-api/#api-fileentry

```suggestion
        file(successCallback: FileCallback, errorCallback?: ErrorCallback): void;
```

HenningCash commented:
@RReverser thanks for the in-depth review. I applied your suggested changes.

> @HenningCash, do you plan to patch this by adding @RReverser as an owner?

done!

HenningCash commented:
Ready to merge
